### PR TITLE
[UKB-To-BIDS] Add error if the data is not found or filtered

### LINKED ISSUE #783: ukb-to-bids converter complains about source_id from zipfile
```
[dlq@beluga2 20227]$ uname -r
3.10.0-1160.76.1.el7.x86_64
```

```
[dlq@beluga2 20227]$ clinica --version
clinica, version 0.7.2
```

```
[dlq@beluga2 20227]$ clinica convert ukb-to-bids 10 ../../Tabular/670041 .
2022-11-08 14:57:33,225:INFO:All clinical data have been found
/home/dlq/.local/lib/python3.8/site-packages/clinica/iotools/converters/ukb_to_bids/ukb_utils.py:86: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only.
  dataframe = pd.concat([dataframe_nifti, dataframe_dicom], 0)
Traceback (most recent call last):
  File "/home/dlq/.local/bin/clinica", line 8, in <module>
    sys.exit(main())
  File "/home/dlq/.local/lib/python3.8/site-packages/clinica/cmdline.py", line 89, in main
    cli()
  File "/home/dlq/.local/lib/python3.8/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/dlq/.local/lib/python3.8/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/dlq/.local/lib/python3.8/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/dlq/.local/lib/python3.8/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/dlq/.local/lib/python3.8/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/dlq/.local/lib/python3.8/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/dlq/.local/lib/python3.8/site-packages/clinica/iotools/converters/ukb_to_bids/ukb_to_bids_cli.py", line 29, in cli
    convert_images(dataset_directory, bids_directory, clinical_data_directory)
  File "/home/dlq/.local/lib/python3.8/site-packages/clinica/iotools/converters/ukb_to_bids/ukb_to_bids.py", line 30, in convert_images
    imaging_data = read_imaging_data(path_to_dataset)
  File "/home/dlq/.local/lib/python3.8/site-packages/clinica/iotools/converters/ukb_to_bids/ukb_utils.py", line 98, in read_imaging_data
    split_zipfile = split_zipfile.astype({"source_id": "int64"})
  File "/home/dlq/.local/lib/python3.8/site-packages/pandas/core/generic.py", line 5887, in astype
    raise KeyError(
KeyError: "Only a column name can be used for the key in a dtype mappings argument. 'source_id' not found in columns."
```

### PR DESCRIPTION
**Pull request by MatthieuJoulot**
Currently, ukb-to-bids converter errors in not so user friendly way if the imaging data is not found or if it is filtered (since we do not convert all of the modalities).

Closes #783.

### CODE REVIEW

File: clinica/iotools/converters/ukb_to_bids/ukb_utils.py [hunk 1]
```diff
@@ -58,6 +58,8 @@ def read_imaging_data(imaging_data_directory: PathLike) -> DataFrame:
 
     import pandas as pd
 
+    from clinica.utils.stream import cprint
```
Comment (NicolasGensollen) [inline clinica/iotools/converters/ukb_to_bids/ukb_utils.py:61]: Could you remove this import if you are not using `cprint` ?

File: clinica/iotools/converters/ukb_to_bids/ukb_utils.py [hunk 2]
```diff
@@ -94,6 +96,10 @@ def read_imaging_data(imaging_data_directory: PathLike) -> DataFrame:
     )
     dataframe = dataframe[filename.isin(file_mod_list)]
     filename = filename[filename.isin(file_mod_list)]
+    if dataframe.empty:
+        raise ValueError(
```
Comment (NicolasGensollen) [inline clinica/iotools/converters/ukb_to_bids/ukb_utils.py:100]: Could you add a unit test for `read_imaging_data` (just covering this corner case) ?
I suppose this error should be raised if you call `read_imaging_data` with an empty folder for example.

File: clinica/iotools/converters/ukb_to_bids/ukb_utils.py [hunk 3]
```diff
@@ -94,6 +96,10 @@ def read_imaging_data(imaging_data_directory: PathLike) -> DataFrame:
...
+            f"Dataset not found, or not handled by Clinica. Please check your data."
```
Comment (NicolasGensollen) [inline clinica/iotools/converters/ukb_to_bids/ukb_utils.py:101]: I would be a bit more specific, maybe something like:

```python
f"No imaging data were found in the provided folder: {imaging_data_directory}. Please check your data."
```

WDYT?

**Review by NicolasGensollen — COMMENTED** [review]
Thanks for looking into this @MatthieuJoulot !

File: clinica/iotools/converters/ukb_to_bids/ukb_utils.py (hunk 3 above)
Comment (MatthieuJoulot) [reply clinica/iotools/converters/ukb_to_bids/ukb_utils.py:101]: If we add the part "not handled by clinica" I agree. Since we do not handle all that is provided.

Comment (NicolasGensollen) [reply clinica/iotools/converters/ukb_to_bids/ukb_utils.py]: > If we add the part "not handled by clinica" I agree. Since we do not handle all that is provided.

Sounds good to me !

File: clinica/iotools/converters/ukb_to_bids/ukb_utils.py (hunk 2 above)
Comment (MatthieuJoulot) [reply clinica/iotools/converters/ukb_to_bids/ukb_utils.py:98]: Yup ! Looking into it, I wonder if it'll work though or how thorough we should be with this unit test. 
Should we just check that it errors right, or should we check that the data generated with a fake image is right as well ? I'm asking because it seems to imply generating complex fake data: fake .nii.gz inside zip files

Comment (MatthieuJoulot) [reply clinica/iotools/converters/ukb_to_bids/ukb_utils.py:98]: Even though it doesn't please me so much, I'd personally go all the way and generate it. WDYT ?

Comment (NicolasGensollen) [reply clinica/iotools/converters/ukb_to_bids/ukb_utils.py:98]: Just check that the error is raised when providing no imaging data (basically only test what this PR is implementing). We'll work on improving the tests later to cover the rest of the function. Does this make sense?

Comment (MatthieuJoulot) [reply clinica/iotools/converters/ukb_to_bids/ukb_utils.py:98]: Alright ! This is much simpler 😄

Comment (NicolasGensollen) [reply clinica/iotools/converters/ukb_to_bids/ukb_utils.py:98]: > Even though it doesn't please me so much, I'd personally go all the way and generate it. WDYT ?

If you are interested in this, why not! 
But I was only suggesting to add a non regression test for this fix.

File: test/unittests/iotools/converters/ukb_to_bids/test_ukb_to_bids_utils.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+import pytest
+
+
+def test_read_imaging_data(tmp_path):
+    import shutil
+
+    from clinica.iotools.converters.ukb_to_bids.ukb_utils import read_imaging_data
+
+    path_to_zip = tmp_path / Path("alt")
```
Comment (NicolasGensollen) [inline test/unittests/iotools/converters/ukb_to_bids/test_ukb_to_bids_utils.py:11]:
```suggestion
    path_to_zip = tmp_path / "alt"
```

File: test/unittests/iotools/converters/ukb_to_bids/test_ukb_to_bids_utils.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
+from pathlib import Path
```
Comment (NicolasGensollen) [inline test/unittests/iotools/converters/ukb_to_bids/test_ukb_to_bids_utils.py:1]:
```suggestion
```

File: clinica/iotools/converters/ukb_to_bids/ukb_utils.py [hunk 6]
```diff
@@ -94,6 +94,10 @@ def read_imaging_data(imaging_data_directory: PathLike) -> DataFrame:
     )
     dataframe = dataframe[filename.isin(file_mod_list)]
     filename = filename[filename.isin(file_mod_list)]
+    if dataframe.empty:
+        raise ValueError(
+            f"No imaging data were found in the provided folder: {imaging_data_directory}, or they are not handled by Clinica. Please check your data."
```
Comment (NicolasGensollen) [inline clinica/iotools/converters/ukb_to_bids/ukb_utils.py:99]: Could you split this line ?

File: test/unittests/iotools/converters/ukb_to_bids/test_ukb_to_bids_utils.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+def test_read_imaging_data(tmp_path):
+    import shutil
+
+    from clinica.iotools.converters.ukb_to_bids.ukb_utils import read_imaging_data
+
+    path_to_zip = tmp_path / Path("alt")
+    shutil.make_archive(path_to_zip, "zip", tmp_path)
+    with pytest.raises(
+        ValueError,
+        match=f"No imaging data were found in the provided folder: {path_to_zip}, or they are not handled by Clinica. Please check your data.",
```
Comment (NicolasGensollen) [inline test/unittests/iotools/converters/ukb_to_bids/test_ukb_to_bids_utils.py:15]: Could you split this line?

**Review by NicolasGensollen — COMMENTED** [review]
Thanks @MatthieuJoulot !
